### DISCUSSION TITLE
Upgrade emqx version 2.2 to version 4.2.8 or above due to vulnerability issue CVE-2021-33175

### DISCUSSION THREAD

**Discussion by boduk25** [emqx/emqx#5010] [Q&A] [answered]
Hi

We are running EMQX broker version 2.2 and planning to upgrade to version 4.2.8 or above.

But before that, I would like to ensure that the latest version won't break our existing application.

Our current EMQX versions are affected by this vulnerability CVE-2021-33175
https://nvd.nist.gov/vuln/detail/CVE-2021-33175
https://www.theregister.com/2021/06/08/mqtt_dos_vulnerabilities/

CVE-2021-33175 - Description
EMQ X Broker versions prior to 4.2.8 are vulnerable to a denial of service attack as a result of excessive memory consumption due to the handling of untrusted inputs. These inputs cause the message broker to consume large amounts of memory, resulting in the application being terminated by the operating system.

Does it require code changes on the application side if I upgrade from 2.2 to the latest version?
What are the breaking changes between version 2.2 and 4.2.8?
Could you advise whether this is safe to proceed?

Thank you so much!

Wendell

**Comment by zmstone** [comment]
converting this to a discussion

**Comment by zmstone** [comment, marked as answer]
Hi @boduk25 
If you are to upgrade, we'd highly recommend 4.3.
It is a big gap from 2.2, there has been however not many changes in MQTT protocol,
so I would not expect any code change in MQTT clients.

The changes in EMQ X are mostly done in:
* Configurations
* Management APIs
* Integrations with external resources (such as auth databases)

**Reply by boduk25** [reply]
Thanks, @zmstone . I appreciate your prompt reply.
